### PR TITLE
fix(web): make dashboard usable on phones and tablets

### LINKED ISSUE #347: Responsive layout: make the dashboard usable on phones and tablets
> 🧑‍💻 **Intermediate Issue** — a complex task spanning multiple modules, with real design decisions to own.
> **Time:** ~25 hours · **Prerequisites:** comfortable navigating this repo (a completed [beginner issue](https://github.com/hiero-hackers/analytics/issues?q=is%3Aissue+state%3Aopen+no%3Aassignee+label%3Abeginner) is the usual route; demonstrated CI/CD proficiency substitutes for workflow-focused issues).
> We expect more than "it works": maintainable code that fits the existing architecture.

## The task

**Problem:** The dashboard is desktop-only by construction. `app.css` contains zero responsive breakpoints — its only `@media` queries are `prefers-color-scheme` (lines 37, 114, 674) — and no Tailwind responsive variants (`sm:`/`md:`/`lg:`) appear anywhere in the TSX. On a 375 px phone:
- `app.css:74-81` — `body { max-width: 1100px; padding: 24px }`: 48 px of fixed padding eats 13 % of the viewport.
- `app.css:283-288` / `:732-738` — `.tablewrap { max-height: 520px }` and `.hipmx-wrap { max-height: 560px }`: nested scroll boxes taller than the viewport that trap the scroll gesture.
- `app.css:106-110` — `td { white-space: nowrap }` on tables up to 11 columns: every table is a sideways scroller.
- `app.css:549-561` — the sticky `.jump` bar wraps to 2-3 rows and permanently occupies a third of the screen.

A public dashboard whose links get opened from Slack and Discord on phones deserves a usable small-screen rendering.

**What done looks like:** The dashboard is *usable* — not pixel-perfect — at 375 px and comfortable at 768 px: padding scales down, scroll containers size to the viewport instead of fixed heights, the jump bar collapses to something compact (or hides on small screens — your call, argued in the approach comment), metric tiles wrap sensibly, and tables remain reachable (horizontal scroll within a table is acceptable and conventional; trapped page-scroll is not). Chart PNGs render at container width. No desktop regressions — the 1100 px experience stays as it is.

**Modules involved / constraints:**
- `web/src/app.css` throughout — decide and state a breakpoint convention (CSS custom media / plain `@media` / Tailwind variants) and apply it consistently; today's zero-breakpoint state means you're setting the precedent the next contributor follows.
- `web/src/components/` — TabBar, MetricTiles, SectionGroups, DataTable are the layout-critical components.
- Verify with real device emulation (DevTools) at 375 px and 768 px, both colour schemes; describe what you checked in the PR. The vitest suite can't see layout — say so honestly rather than writing tests that assert class names.
- Scope: layout only. The separate sticky-bar anchor-offset and jump-link bugs are tracked on their own issues.

## How to work on this

1. **Claim it:** comment `/assign` and wait to be assigned — unassigned PRs are closed automatically.
2. **Get a plan:** once assigned, comment `@coderabbitai plan` for a draft plan, then do your own investigation — [docs/architecture.md](https://github.com/hiero-hackers/analytics/blob/main/docs/architecture.md) maps the layers and their rules.
3. **Propose your approach as a comment before coding.** A paragraph is enough; early feedback here routinely saves days of rework.

**🤖 AI:** tools are welcome; verified work is required — you can explain every line and defend every design choice. See the [AI policy](https://github.com/hiero-hackers/analytics/blob/main/CONTRIBUTING.md#ai-policy). Fully automated bot PRs are closed.

**Worth knowing about this repo before you design:**

- The layer rules in [docs/architecture.md](https://github.com/hiero-hackers/analytics/blob/main/docs/architecture.md) are strict — review will hold your solution to them.
- Tests mirror src (`tests/<pkg>/test_<module>.py`), and the output-contract test pins the pipeline output surface — if your change adds or renames outputs, update the contract deliberately.

**Before opening your PR:**

- [ ] I proposed my approach on this issue and incorporated any feedback
- [ ] The solution fits the existing architecture and layer rules, and is clear enough for others to debug without me
- [ ] Tests cover the happy path, edge cases, and error handling ([testing guide](https://github.com/hiero-hackers/analytics/blob/main/CONTRIBUTING.md#testing))
- [ ] I reviewed my own diff line by line; scope is limited to this issue
- [ ] Workflow checks pass — CI green, signed commits, linked issue

**Stuck?** Comment here with what you've tried — see [getting help](https://github.com/hiero-hackers/analytics/blob/main/CONTRIBUTING.md#getting-help).

### PR DESCRIPTION
**Pull request by darshit2308**
Fixes #347

The dashboard had zero responsive breakpoints. On a 375 px phone, 48 px of fixed body padding consumed 13% of the viewport, the table scroll containers were taller than the visible screen (trapping the page-scroll gesture), and the section header crammed description text and action buttons side-by-side with no room to breathe.

### What changed

**`web/src/app.css` only** : no TSX components touched.

- **Body padding** scales down: 24px on desktop → 16px on tablets (≤767px)  -> 12px on phones (≤599px). `h1` shrinks from 26px to 22px on phones.
- **Table scroll containers** — `.tablewrap` and `.hipmx-wrap` now use `min(520px, 60dvh)` and `min(560px, 60dvh)` instead of fixed pixel heights. On a 667px phone this gives the container a ~400px cap, leaving the page scroll gesture always reachable above and below the table.
- **Section header** — at ≤479px, description stacks above the action buttons instead of sitting side-by-side.
- **Jump bar** — already scrolls horizontally (`flex-wrap: nowrap; overflow-x: auto`), so no structural change needed. The "Jump to" label is hidden at ≤479px to give the buttons more room.
- **Glossary grid** — the fixed 150px definition column shrinks to`minmax(0, 120px)` on narrow cards so it doesn't eat more than half the available width.
- **Card padding** — tightens to `12px 14px` at ≤479px, recovering 8px of horizontal room per side.

### What I left alone (and why)

- `td { white-space: nowrap }` — the issue says horizontal table scroll is acceptable. The scroll trap came from the fixed max-height, not the columns.
- The chart gallery uses `repeat(auto-fit, minmax(320px, 1fr))` which already collapses to one column on a 375px phone. No change needed.
- MetricTiles, PeriodTabs, TabBar — already responsive via Tailwind utilities and `flex-wrap: wrap`.

### Breakpoint convention

This is the first responsive addition to this file, so I documented the convention in a comment block at the top of the responsive section: plain `@media` queries (no CSS custom-media, no Tailwind `sm:`/`md:` in the component-class layer), two breakpoints — tablet ≤767px, phone ≤599px — so the next contributor has a clear pattern to follow.

### Testing

- `oxlint` — 0 errors (5 pre-existing warnings, none in the edited file)
- `prettier --check` — clean
- `vitest` — 62/62 tests pass (CSS layout can't be tested in jsdom; noted honestly rather than writing tests that assert class names)

Verified manually in Chrome DevTools device emulation at 375px (iPhone SE), 768px (iPad Mini), and 1440px (desktop), both light and dark colour schemes. Desktop layout is unchanged.

### CODE REVIEW

File: web/src/app.css [hunk 1]
```diff
@@ -450,9 +485,18 @@
...
+     `min(520px, 60dvh)` for full support. `dvh` (dynamic viewport height)
+     accounts for collapsible browser chrome; `60dvh` leaves at least 40 % of
+     the screen for content above and below the table so the page-scroll
+     gesture cannot be trapped. */
   .tablewrap {
     overflow: auto;
-    max-height: 520px;
+    max-height: 520px; /* no min() support: fixed cap */
+    max-height: min(520px, 60vh); /* min() + vh, no dvh: viewport-relative */
+    max-height: min(520px, 60dvh); /* full support: dvh accounts for browser chrome */
```
Comment (phillip-nyinomujuni) [inline web/src/app.css:499]: Did you actually test this in a browser that has min() but not `dvh`? Old Safari is the case I'd worry about. Should fall through to the `vh` line, just want to know it's confirmed and not just assumed.

File: web/src/app.css [hunk 2]
```diff
@@ -899,9 +943,12 @@
      on a dark page. The heat buckets read from --heat-*, which inverts for
      dark mode (see the palette); the matrix legend renders the same classes,
      so cells and legend cannot drift. */
+  /* Same three-level cascade as .tablewrap above (560 px / 60vh / 60dvh). */
   .hipmx-wrap {
     overflow: auto;
-    max-height: 560px;
+    max-height: 560px; /* no min() support: fixed cap */
+    max-height: min(560px, 60vh); /* min() + vh, no dvh: viewport-relative */
+    max-height: min(560px, 60dvh); /* full support: dvh accounts for browser chrome */
```
Comment (phillip-nyinomujuni) [inline web/src/app.css:951]: Same 3-line cascade as `.tablewrap`, just different numbers. Since the comment above says this sets the precedent for everyone else, can we pull it into one shared thing instead of copy-pasting it? Otherwise one of these drifts eventually.

File: web/src/app.css [hunk 3]
```diff
@@ -1275,4 +1322,57 @@
...
+    .jlabel {
+      position: absolute;
+      width: 1px;
+      height: 1px;
+      padding: 0;
+      margin: -1px;
+      overflow: hidden;
+      clip: rect(0, 0, 0, 0);
+      white-space: nowrap;
+      border: 0;
```
Comment (phillip-nyinomujuni) [inline web/src/app.css:1362]: Did you check this with a screen reader? Want to make sure "Jump to" actually gets announced before the button list at this width, not just that it's visually hidden.

**Review by phillip-nyinomujuni — COMMENTED** [review]
Code-wise this is solid. Main things I want confirmed before merge: the `dvh` fallback tested on a real old-Safari-type browser, and the `.jlabel` label actually announced by a screen reader. The `.tablewrap`/`.hipmx-wrap` duplication is a nit, can be a follow-up.

File: web/src/app.css (hunk 1 above)
Comment (darshit2308) [reply web/src/app.css:499]: The cascade is guaranteed by CSS parsing rules, not an assumption. When a browser encounters an unknown unit (dvh), the entire min(520px, 60dvh) declaration is dropped and it falls back to the last valid max-height it saw, which is min(520px, 60vh). 
Old Safari (11.1–15.3) supports min() and vh since 2018, so it correctly lands on the vh line.

File: web/src/app.css (hunk 2 above)
Comment (darshit2308) [reply web/src/app.css:951]: yep, it is duplication. This can be deduplicated cleanly with a CSS custom property per element and a shared grouped selector.
This can be solved in a follow up issue or PR.

File: web/src/app.css (hunk 3 above)
Comment (darshit2308) [reply web/src/app.css:1362]: The pattern used is the industry-standard visually-hidden technique, same as Bootstrap's .visually-hidden and Tailwind's sr-only. It keeps "Jump to" in the accessibility tree (element is still in the DOM, no display:none, no aria-hidden), so screen readers will announce it before the button list.

**Review by danielmarv — APPROVED** [review]
